substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/SubstrateJVM.java

```diff
@@ -325,8 +347,9 @@ public void setMethodSamplingInterval(@SuppressWarnings("unused") long type, @Su
     }
 
     /** See {@link JVM#setSampleThreads}. */
-    public void setSampleThreads(@SuppressWarnings("unused") boolean sampleThreads) {
-        throw new IllegalStateException("JFR Thread sampling is currently not supported.");
+    public void setSampleThreads(boolean sampleThreads) {
+        setEnabled(JfrEvents.ExecutionSample.getId(), sampleThreads);
+        setEnabled(JfrEvents.NativeMethodSample.getId(), sampleThreads);
```

Comment:
This PR is the first in a series of PRs. The second PR (I'll push it after we manage to merge these changes) will introduce support for stack traces. I split those PRs into smaller ones because of testing and reviewing.